src/main/java/bl4ckscor3/mod/globalxp/openmods/utils/EnchantmentUtils.java

```diff
@@ -23,15 +23,15 @@ private EnchantmentUtils() {}
 	 * @param player The player to get the XP of
 	 * @return The amount of XP the given player has
 	 */
-	public static int getPlayerXP(Player player) {
-		return (int) (EnchantmentUtils.getExperienceForLevel(player.experienceLevel) + (player.experienceProgress * player.getXpNeededForNextLevel()));
+	public static long getPlayerXP(Player player) {
+		return (long) (EnchantmentUtils.getExperienceForLevel(player.experienceLevel) + (player.experienceProgress * player.getXpNeededForNextLevel()));
 	}
 
-	public static void addPlayerXP(Player player, int amount) {
-		int experience = getPlayerXP(player) + amount;
-		int expForLevel;
+	public static void addPlayerXP(Player player, long amount) {
+		long experience = getPlayerXP(player) + amount;
+		long expForLevel;
 
-		player.totalExperience = experience;
+		player.totalExperience = (int)Math.min(Integer.MAX_VALUE, experience);

```

Comment:
From a quick glance, I think XP loss can occur here when the player is removing an amount of XP from the block that would result in them having more XP than the integer limit. I have to properly test this when I have more time to look at my mods.

Reply:
This should be correct, however the amount of XP levels for that to happen is 238,609,312 according to the minecraft wiki https://minecraft.wiki/w/Experience#Useful_numbers

Reply:
If you have more than the integer limit of XP on your player, Minecraft works just fine even though this value gets stuck at the max integer value. (Testable by setting your XP to 0 points and levels, then running `/xp add @s 2147483647` twice). So maybe this is a non-issue, but still worth looking into.

Reply:
It does break at some point after 47721862 (which is not anywhere near the vanilla maximum levels), so I am going to debug it.